### PR TITLE
Add aro-e2e workflow

### PR DESCRIPTION
**Pull request by slintes**
Needs access to RH network, configure test according to https://docs.ci.openshift.org/docs/architecture/ci-operator/#allowing-tests-to-access-red-hat-intranet

Test also needs https://github.com/openshift/ci-tools/pull/4318 and https://github.com/openshift/release/pull/56644

### CODE REVIEW

File: ci-operator/step-registry/aro-e2e/classic/aro-e2e-classic-workflow.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    post:
+      - ref: aro-e2e-classic-cluster-delete
+      - ref: aro-e2e-classic-bootstrap-delete
+    env:
+      AZURE_SUBSCRIPTION_ID: "fe16a035-e540-4ab7-80d9-373fa9a3d6ae"
+      AZURE_LOCATION: "eastus"
+      RP_ENDPOINT: "https://4.255.86.56:443"
+      ARO_VERSION: "4.14.33"
+      ARO_VERSION_OPENSHIFT_PULLSPEC: "quay.io/openshift-release-dev/ocp-release@sha256:40ba8c540d16a97a6b629232a5a16da64fca655c4b83b734aa28415b5c708acf"
+      ARO_VERSION_INSTALLER_PULLSPEC: "ocpciarotest.azurecr.io/aro-installer:release-4.14"
```
Comment (stbenjam) [inline ci-operator/step-registry/aro-e2e/classic/aro-e2e-classic-workflow.yaml:19]: Is it not possible to use an arbitrary release payload yet?

Comment (slintes) [reply ci-operator/step-registry/aro-e2e/classic/aro-e2e-classic-workflow.yaml:19]: I want to keep it simple in the first iteration, but in a follow up the  plan is to get this from the `releases` config in the actual test.

**Comment by slintes** [comment]
/hold

I'm going to create a test job which uses this, so we can run rehearsals on this PR

**Comment by slintes** [comment]
/hold cancel

I can't create a test job using this workflow without having this workflow merged 🤷🏼‍♂️ 

New plan:
- merge this without knowing if it works
- create and merge a test job (I would be suprised if the workflow works out of the box)
- create a follow PR for this workflow which fixes issues using rehearsal jobs

**Review by tsatam — APPROVED** [review]

**Review by tsatam — APPROVED** [review]

**Pull request merged**